.github/workflows/scala.yml

```diff
@@ -17,54 +17,88 @@ jobs:
+        id: date
+        run: echo "::set-output name=DATE::$(date +'%Y-%m-%d')"
+
+      - name: Checkout ViperServer
+        uses: actions/checkout@v2
+        with:
+          path: viperServer
+      - name: Checkout Silver
+        uses: actions/checkout@v2
+        with:
+          repository: viperproject/silver
+          path: silver
+      - name: Checkout Silicon
+        uses: actions/checkout@v2
+        with:
+          repository: viperproject/silicon
+          path: silicon
+      - name: Checkout Carbon
+        uses: actions/checkout@v2
+        with:
+          repository: viperproject/carbon
+          path: carbon
     
-    - name: Set up JDK 1.8
-      uses: actions/setup-java@v1
-      with:
-        java-version: 1.11
+      - name: Symbolically link Silver to Carbon
+        run: cd carbon; ln --symbolic ../silver
+      - name: Symbolically link Silver to Silicon
+        run: cd silicon; ln --symbolic ../silver
+      - name: Symbolically link Silver to ViperServer
+        run: cd viperServer; ln --symbolic ../silver
+      - name: Symbolically link Carbon to ViperServer
+        run: cd viperServer; ln --symbolic ../carbon
+      - name: Symbolically link Silicon to ViperServer
+        run: cd viperServer; ln --symbolic ../silicon
+
+      - name: Install Z3
+        env: 
+          Z3_VERSION: 4.8.9
+          Z3_PLATFORM: x64
+          Z3_OS: ubuntu-16.04
+        run: |
+          export Z3=z3-$Z3_VERSION-$Z3_PLATFORM-$Z3_OS; 
+          curl -J -L https://github.com/Z3Prover/z3/releases/download/z3-$Z3_VERSION/$Z3.zip > z3.zip; 
+          unzip z3.zip; 
+          sudo ln -s $(pwd)/$Z3/bin/z3 /usr/bin/z3
 
-    - name: Install SBT
-      run: echo "deb https://dl.bintray.com/sbt/debian /" | sudo tee -a /etc/apt/sources.list.d/sbt.list; curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | sudo apt-key add; sudo apt-get update; sudo apt-get install sbt
+      - name: Set up JDK 15
+        uses: actions/setup-java@v1
+        with:
+          java-version: '15.0.1'
+          architecture: x64
+
+      - name: Install SBT
+        run: |
+          echo "deb https://dl.bintray.com/sbt/debian /" | sudo tee -a /etc/apt/sources.list.d/sbt.list; 
+          curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | sudo apt-key add; 
+          sudo apt-get update; sudo apt-get install sbt
+    
+      - name: test project
+        run: |
+          cd viperServer; 
+          export Z3_EXE="/usr/bin/z3"; 
+          env > print_dir.txt; 
+          sbt test
 
-    - name: Assemble project
-      run: cd viperServer; sbt assembly
-        
-    - name: test project
-      run: cd viperServer; export Z3_EXE="/usr/bin/z3"; env > print_dir.txt; sbt test
+      - name: Assemble the fat JAR file
+        run: |
+          cd viperServer; 
+          sbt assembly
       
+      - name: Publish Nightly Artifact
```

Comment:
I think publishing should only happen if this step is run as part of a cron job (and on master) instead for every single push and pull request. In addition, I would split the job into multiple jobs and communicate via uploading/downloading artifacts

Reply:
I agree with the suggestion. Can we tune the CI script after merging this PR, though?

Reply:
Sure, but please add an issue so that we do not forget about it

Reply:
See #27